parsl/dataflow/future_resolution.py

```diff
@@ -55,3 +56,28 @@ def unwrap(v):
 
     type_ = type(iterable)
     return type_(map(unwrap, iterable))
+
+
+@traverse_to_gather.register(Mapping)
+def _(dictionary):
+    futures = []
+    for key, value in dictionary.items():
+        if isinstance(key, Future):
```

Comment:
you should be able to call traverse_to_gather here without an if statement and it will recursively find the futures for you. at least thats the intention.